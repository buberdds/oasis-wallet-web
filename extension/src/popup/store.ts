import { Store, applyMiddleware } from 'webext-redux'
import createSagaMiddleware from 'redux-saga'

export default function configureAppStore() {
  const store = new Store()
  const sagaMiddleware = createSagaMiddleware({})
  const middlewares = [sagaMiddleware]
  const storeWithMiddleware = applyMiddleware(store, ...middlewares)
  return storeWithMiddleware
}
