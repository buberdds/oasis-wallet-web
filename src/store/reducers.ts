/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from '@reduxjs/toolkit'
import createWalletReducer from 'app/pages/CreateWalletPage/slice'
import openWalletReducer from 'app/pages/OpenWalletPage/slice'
import accountReducer from 'app/state/account'
import fatalErrorReducer from 'app/state/fatalerror'
import ledgerReducer from 'app/state/ledger'
import networkReducer from 'app/state/network'
import stakingReducer from 'app/state/staking'
import transactionReducer from 'app/state/transaction'
import walletReducer from 'app/state/wallet'
import themeReducer from 'styles/theme/slice'

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(injectedReducers = {}) {
  // Initially we don't have any injectedReducers, so returning identity function to avoid the error
  const rootReducer = combineReducers({
    ...injectedReducers,
    account: accountReducer,
    createWallet: createWalletReducer,
    fatalError: fatalErrorReducer,
    ledger: ledgerReducer,
    network: networkReducer,
    openWallet: openWalletReducer,
    staking: stakingReducer,
    theme: themeReducer,
    transaction: transactionReducer,
    wallet: walletReducer,
  })

  return rootReducer
}
