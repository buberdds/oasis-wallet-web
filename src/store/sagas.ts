import { all } from 'typed-redux-saga'

import { accountSaga } from 'app/state/account/saga'
import { networkSaga } from 'app/state/network/saga'
import { ledgerSaga } from 'app/state/ledger/saga'
import { stakingSaga } from 'app/state/staking/saga'
import { transactionSaga } from 'app/state/transaction/saga'
import { rootWalletSaga } from 'app/state/wallet/saga'

export default function* rootSagas() {
  yield all([accountSaga(), networkSaga(), ledgerSaga(), stakingSaga(), transactionSaga(), rootWalletSaga()])
}
