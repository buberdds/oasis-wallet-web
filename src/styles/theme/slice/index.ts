import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { useInjectReducer } from 'utils/redux-injectors'

import { getThemeFromStorage, saveTheme } from '../utils'
import { ThemeState } from './types'

export const initialState: ThemeState = {
  selected: getThemeFromStorage() || 'system',
}

const slice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    changeTheme(state, action: PayloadAction<'dark' | 'light'>) {
      saveTheme(action.payload)
      state.selected = action.payload
    },
  },
})

// export const { actions: themeActions, reducer } = slice

// export const useThemeSlice = () => {
//   useInjectReducer({ key: slice.name, reducer: slice.reducer })
//   return { actions: slice.actions }
// }
console.log(slice)
export const actions = slice.actions
export default slice.reducer
