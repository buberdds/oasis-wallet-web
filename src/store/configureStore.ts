import { configureStore } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'

import { createReducer } from './reducers'
import rootSagas from './sagas'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.toString(),
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware]

  const store = configureStore({
    reducer: createReducer(),
    middleware: getDefaultMiddleware => getDefaultMiddleware().concat(middlewares),
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production',
    preloadedState: state,
  })

  sagaMiddleware.run(rootSagas)

  return store
}
