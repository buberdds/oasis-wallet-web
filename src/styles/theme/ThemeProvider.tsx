import { grommet, Grommet, ThemeType } from 'grommet'
import { deepMerge } from 'grommet/utils'
import * as React from 'react'
import { createTheme, IDataTableStyles, ITheme } from 'react-data-table-component'
import { useSelector } from 'react-redux'

import { selectTheme } from './slice/selectors'

/**
 * React-data-table by default sets its own background and text colors
 * we make sure that they do not override grommet's
 */
createTheme('blank', {
  background: { default: 'false' },
  highlightOnHover: {
    default: '#88888833',
    text: 'false',
  },
  text: {
    primary: 'false',
    secondary: 'false',
    disabled: 'false',
  },
  sortFocus: {
    default: 'false',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>)

export const dataTableStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '18px',
    },
  },
  rows: {
    style: {
      fontSize: '16x',
    },
  },
}

const customTheme: ThemeType = {
  button: {
    primary: {
      background: {
        dark: 'red',
        light: 'green',
      },
    },
    // default: {
    //   background: {
    //     dark: '#7D4CDB',
    //     light: '#7D4CDB',
    //   },
    // },
  },
  global: {
    colors: {
      oasisIndigo: '#310081',
      oasisMinty: '#4CD4A9',
      oasisLightGray: '#ececec',
      brand: {
        light: '#0092f6',
        dark: '#0092f6bb',
      },
      oasisBlue2: '#4db3f9',
      oasisBlue3: '#26a2f8',
      'status-ok': '#4cd4a9',
      'status-error': '#ff4212',
      'status-warning': '#f4ce4b',
      'background-oasis-blue': {
        dark: '#0f477b',
        light: '#0092f6',
      },
      neutral: {
        dark: '#310081FF',
        light: '#310081FF',
      },
      'neutral-2': {
        dark: '#0092f6bb',
        light: '#0092f6bb',
      },
      'background-back': {
        dark: '#1A1A2e',
        light: '#EFEFEF',
      },
      'background-front': {
        dark: '#16213e',
        light: '#FFFFFF',
      },
      'background-front-lighter': {
        dark: '#222938',
        light: '#FFFFFF',
      },
      'background-front-border': {
        dark: '#111111',
        light: '#EDEDED',
      },
      'background-contrast': {
        dark: '#FFFFFF08',
        light: '#11111108',
      },
      'background-contrast-2': {
        dark: '#FFFFFF33',
        light: '#11111133',
      },
      'background-custom': {
        dark: '#0E5265',
        light: '#00C8FF',
      },
      'component-toolbar': {
        dark: '#0f346088',
        light: '#26a2f844',
      },
      'component-sidebar': {
        dark: '#16213e',
        light: '#fafafa',
      },
    },
    font: {
      family: 'Rubik',
    },
    opacity: {
      weak: 0.25,
    },
  },
}
export const ThemeProvider = (props: { children: React.ReactChild }) => {
  const theme = deepMerge(grommet, customTheme)
  const mode = useSelector(selectTheme)

  return (
    <Grommet theme={theme} themeMode={mode} style={{ minHeight: '100vh' }}>
      {React.Children.only(props.children)}
    </Grommet>
  )
}
