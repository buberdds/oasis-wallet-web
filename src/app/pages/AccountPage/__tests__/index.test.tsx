import { render } from '@testing-library/react'
import * as React from 'react'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { createMemoryHistory } from 'history'
import { configureAppStore } from 'store/configureStore'

import { AccountPage } from '..'

const renderPage = (store: any, history: any) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <MemoryRouter>
          <AccountPage />
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  )

describe('<AccountPage  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  it.skip('should match snapshot', () => {
    const history = createMemoryHistory()
    history.push('/account/account_address')
    const page = renderPage(store, history)
    expect(page.container.firstChild).toMatchSnapshot()
  })
})
