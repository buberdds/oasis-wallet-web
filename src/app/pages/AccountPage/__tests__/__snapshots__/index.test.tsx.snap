// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountPage  /> should match snapshot 1`] = `
.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 18px;
  height: 18px;
  fill: #666666;
  stroke: #666666;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c10 *[stroke*="#"],
.c10 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*="#"],
.c10 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: rgba(76,212,169,0.25);
  border-left: solid 3px #4cd4a9;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  border: solid 1px #EDEDED;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 5px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  padding: 12px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-right: 12px;
  border-radius: 5px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0092f6;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20px;
  padding-left: 6px;
  padding-right: 6px;
  border-radius: 24px;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 300px;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-basis: 25%;
  -ms-flex-preferred-size: 25%;
  flex-basis: 25%;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 24px;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-basis: 75%;
  -ms-flex-preferred-size: 75%;
  flex-basis: 75%;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c32 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  border: solid 1px #EDEDED;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 48px;
  border-radius: 5px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100%;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 12px;
  margin-bottom: 12px;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c26 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 24px;
}

.c29 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c11 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
  word-break: break-word;
}

.c14 {
  margin-left: 3px;
  font-size: 12px;
  line-height: 18px;
  color: #a3a3a3;
  font-weight: 600;
}

.c13 {
  font-size: 18px;
  line-height: 24px;
}

.c18 {
  font-size: 14px;
  line-height: 20px;
  font-weight: bold;
}

.c22 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible) > circle,
.c9:focus:not(:focus-visible) > ellipse,
.c9:focus:not(:focus-visible) > line,
.c9:focus:not(:focus-visible) > path,
.c9:focus:not(:focus-visible) > polygon,
.c9:focus:not(:focus-visible) > polyline,
.c9:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c28 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0092f6;
  color: #f8f8f8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c28:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c28:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c28:focus > circle,
.c28:focus > ellipse,
.c28:focus > line,
.c28:focus > path,
.c28:focus > polygon,
.c28:focus > polyline,
.c28:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c28:focus::-moz-focus-inner {
  border: 0;
}

.c28:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c28:focus:not(:focus-visible) > circle,
.c28:focus:not(:focus-visible) > ellipse,
.c28:focus:not(:focus-visible) > line,
.c28:focus:not(:focus-visible) > path,
.c28:focus:not(:focus-visible) > polygon,
.c28:focus:not(:focus-visible) > polyline,
.c28:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c28:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c33 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c25 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c25::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c25::-moz-placeholder {
  color: #AAAAAA;
}

.c25:-ms-input-placeholder {
  color: #AAAAAA;
}

.c25::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c25::-moz-focus-inner {
  border: none;
  outline: none;
}

.c25:-moz-placeholder,
.c25::-moz-placeholder {
  opacity: 1;
}

.c24 {
  position: relative;
  width: 100%;
}

.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-top: solid 1px #EDEDED;
  margin: 6px 0 0;
  padding: 12px 12px 6px;
}

.c12 dt,
.c12 dd {
  margin: 0;
}

.c12 dt:first-of-type,
.c12 dd:first-of-type {
  font-weight: bold;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  padding: 12px;
}

.c16:hover {
  background-color: #11111108;
}

.c16.active {
  background-color: #EFEFEF;
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-left: solid 3px #4cd4a9;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding-top: 3px;
    padding-bottom: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border: solid 1px #EDEDED;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {
  .c17 {
    border-radius: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c19 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:768px) {
  .c21 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c23 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c27 {
    margin-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c31 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c32 {
    border: solid 1px #EDEDED;
  }
}

@media only screen and (max-width:768px) {
  .c32 {
    padding: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    margin-top: 6px;
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c26 {
    height: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c29 {
    width: auto;
    height: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c33 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (min-width:768px) {
  .c12 dt {
    width: 40%;
  }

  .c12 dd {
    width: 60%;
  }

  .c12 dt:not(:last-of-type),
  .c12 dd:not(:last-of-type) {
    margin-bottom: 6px;
  }

  .c12 dt:first-of-type,
  .c12 dd:first-of-type {
    font-size: 22px;
    line-height: 28px;
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c12 dt {
    width: 30%;
  }

  .c12 dd {
    width: 70%;
    text-align: right;
  }

  .c12 dt,
  .c12 dd {
    font-size: 16px;
  }

  .c12 dt:not(:last-of-type),
  .c12 dd:not(:last-of-type) {
    margin-bottom: 3px;
  }

  .c12 dt:first-of-type,
  .c12 dd:first-of-type {
    font-size: 18px;
    line-height: 24px;
    margin-bottom: 6px;
  }
}

<div
  class="c0"
  style="min-height: 100vh;"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          This is your account.
        </span>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <button
              class="c9"
              type="button"
            >
              <svg
                aria-label="Copy"
                class="c10"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 15h8-8zm0-4h10H9zm0-4h4-4zm7-6v6h6M6 5H2v18h16v-4m4 0H6V1h11l5 5v13z"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </button>
            <span
              class="c11"
            >
              oasis1 qz0k 5q8v jqvu 4s4n wxyj 406y lnfl kc4v rcjg huwk
            </span>
          </div>
          <dl
            class="c12"
            data-testid="account-balance-summary"
          >
            <dt>
              <span
                class="c13"
              >
                Total
              </span>
            </dt>
            <dd
              data-testid="account-balance-total"
            >
              <span>
                100.000001111
                <span
                  class="c14"
                />
              </span>
            </dd>
            <dt>
              Available
            </dt>
            <dd>
              <span>
                100.0
                <span
                  class="c14"
                />
              </span>
            </dd>
            <dt>
               
              Staked
            </dt>
            <dd>
              <span>
                0.000001111
                <span
                  class="c14"
                />
              </span>
            </dd>
            <dt>
              Debonding
            </dt>
            <dd>
              <span>
                0.0
                <span
                  class="c14"
                />
              </span>
            </dd>
          </dl>
        </div>
      </div>
    </div>
    <nav
      class="c15"
    >
      <a
        aria-current="page"
        class="c16 active"
        href="/account/oasis1qz0k5q8vjqvu4s4nwxyj406ylnflkc4vrcjghuwk"
      >
        <span
          class="c13"
        >
          Transactions
        </span>
      </a>
      <a
        class="c16"
        href="/account/oasis1qz0k5q8vjqvu4s4nwxyj406ylnflkc4vrcjghuwk/active-delegations"
      >
        <span
          class="c13"
        >
          Delegations
        </span>
        <div
          class="c17"
          style="position: absolute; top: -3px; right: -5px; z-index: 2;"
        >
          <span
            class="c18"
          >
            1
          </span>
        </div>
      </a>
      <a
        class="c16"
        href="/account/oasis1qz0k5q8vjqvu4s4nwxyj406ylnflkc4vrcjghuwk/debonding-delegations"
      >
        <span
          class="c13"
        >
          Debonding
        </span>
      </a>
    </nav>
    <div
      class="c19"
    >
      <div
        class="c20"
      >
        <div
          class="c5"
        >
          <form>
            <div
              class="c21"
            >
              <div
                class="c2 "
              >
                <label
                  class="c22"
                  for="recipient-id"
                >
                  Recipient
                </label>
                <div
                  class="c23 FormField__FormFieldContentBox-sc-m9hood-1"
                >
                  <div
                    class="c24"
                  >
                    <input
                      autocomplete="off"
                      class="c25"
                      id="recipient-id"
                      name="recipient"
                      placeholder="Enter an address"
                      required=""
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="c26"
              />
              <div
                class="c2 "
              >
                <label
                  class="c22"
                  for="amount-id"
                >
                  Amount
                </label>
                <div
                  class="c23 FormField__FormFieldContentBox-sc-m9hood-1"
                >
                  <div
                    class="c24"
                  >
                    <input
                      autocomplete="off"
                      class="c25"
                      id="amount-id"
                      min="0"
                      name="amount"
                      placeholder="0"
                      required=""
                      step="any"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="c26"
              />
              <div
                class="c27"
              >
                <button
                  class="c28"
                  type="submit"
                >
                  Send
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="c29"
      />
      <div
        class="c30"
      >
        <div
          class="c31"
        >
          <div
            class="c32"
          >
            <h3
              class="c33"
            >
              No transactions found.
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AccountPage  /> with missing delegations 1`] = `"Total-Available100.0 Staked-Debonding-"`;

exports[`<AccountPage  /> with missing delegations 2`] = `"TransactionsDelegationsDebonding"`;
